src/suites/cts/copyImageBitmapToTexture.spec.ts

```diff
@@ -146,3 +146,97 @@ g.test('from ImageData', async t => {
     poptions('height', [1, 2, 4, 15, 255, 256])
   )
 );
+
+g.test('from canvas', async t => {
+  const { width, height } = t.params;
+
+  // The texture format is rgba8uint, so the bytes per pixel is 4.
+  const bytesPerPixel = 4;
+
+  // Generate original data.
+  const imagePixels = new Uint8ClampedArray(bytesPerPixel * width * height);
+  for (let i = 0; i < width * height * bytesPerPixel; ++i) {
+    imagePixels[i] = i % 256;
+  }
+
+  const imageData = new ImageData(imagePixels, width, height);
+
+  // CTS works on workder thread sometime where doucment
+  // element is not available. In such situation, offscreenCanvas
+  // can be the candidate. But not all browsers support creating
+  // 2d context on offscreenCanvas. In this situation, the case will
+  // be skipped.
```

Comment:
I believe Safari doesn't support OffscreenCanvas at all, so you'll have to check for `typeof OffscreenCanvas === 'undefined'` too.

(This will fail on Safari anyway because they do not have ImageBitmap. But we should still skip the test if OffscreenCanvas isn't supported.)

Reply:
Done.